src/DepFile.cpp

```diff
@@ -69,51 +135,71 @@ static bool sParseMakeDepFile(FileID inDepFileID, StringView inDepFileContent, s
 	}
 
 	dep_file_content = dep_file_content.substr(deps_start + cDepStartMarker.size());
-
+	
 	while (!dep_file_content.empty())
 	{
 		// Skip white space before the path.
 		while (!dep_file_content.empty() && sIsSpace(dep_file_content.front()))
 			dep_file_content = dep_file_content.substr(1);
 
 		// Lines generally end with space + backslash + linefeed.
+		constexpr StringView cWindowsLineEnd  = " \\\r\n";
 		constexpr StringView cLineEnd  = " \\\n";
-		size_t path_end  = dep_file_content.find(cLineEnd);
-		size_t next_path = path_end + cLineEnd.size();
+		size_t path_end  = dep_file_content.find(cWindowsLineEnd);
+		size_t next_path = path_end + cWindowsLineEnd.size();
+
+		// Match first on Windows' CRLF before matching on LF.
+		if (path_end == StringView::npos)
+		{
+			path_end  = dep_file_content.find(cLineEnd);
```

Comment:
alternatively you could search for \n only, but then check if the previous char is \r and do `path_end--` if yes (it avoids doing the find twice, but feel free to leave it like that, I'm nitpicking :))

Reply:
I currently have one commit yet to push trying to change that part of the algorithm but it turns the code into a chain of if/else if that makes the code a bit more fragile. For instance, it asserts on remaining spaces a line (`: file \n`). I have to investigate in a better solution that doesn't just involve an additional space trim before sending it.

Also I'm not sure about checking only the presence of `\r`. My current local version does incremental search to look for:
1. `\n`
2. `\r`
3. `\`

`\r` and `\` being optional here, each check shifts `path_end` backwards. This should handle all the possible line returns, assuming we don't end up with LFCR or LF alone.

I have to ask if we should consider a non-escaped newline as the end of the process, as it *should* be in the context of Make rules? if so, the involved logic might a bit more complex to handle the different combinations (or I'm completely missing an obvious solution here)

In the meantime, I don't mind if you merge the PR as-is (with the style fixup I sent up a bit before writing this comment.